packages/rum/src/rum.ts

```diff
@@ -205,29 +205,35 @@ export function startRum(
   startViewCollection(location, lifeCycle)
 
   return {
-    addRumGlobalContext: monitor((key: string, value: ContextValue) => {
-      globalContext[key] = value
-    }),
-    addUserAction: monitor((name: string, context?: Context) => {
-      lifeCycle.notify(LifeCycleEventType.CUSTOM_ACTION_COLLECTED, { context, name, type: UserActionType.CUSTOM })
-    }),
-    getInternalContext: monitor(
-      (startTime?: number): InternalContext => {
-        return (withSnakeCaseKeys(deepMerge(
-          { applicationId },
-          parentContexts.findView(startTime),
-          parentContexts.findAction(startTime)
-        ) as Context) as unknown) as InternalContext
-      }
-    ),
-    setRumGlobalContext: monitor((context: Context) => {
-      globalContext = context
-    }),
+    globalApi: {
+      addRumGlobalContext: monitor((key: string, value: ContextValue) => {
+        globalContext[key] = value
+      }),
+      addUserAction: monitor((name: string, context?: Context) => {
+        lifeCycle.notify(LifeCycleEventType.CUSTOM_ACTION_COLLECTED, { context, name, type: UserActionType.CUSTOM })
+      }),
+      getInternalContext: monitor(
+        (startTime?: number): InternalContext => {
+          return (withSnakeCaseKeys(deepMerge(
+            { applicationId },
+            parentContexts.findView(startTime),
+            parentContexts.findAction(startTime)
+          ) as Context) as unknown) as InternalContext
+        }
+      ),
+      setRumGlobalContext: monitor((context: Context) => {
+        globalContext = context
+      }),
+    },
+    stop: () => {
+      batch.stop()
```

Comment:
I feel that we should rather stop collecting stuff (stop the view collection etc.) instead of telling the batch to ignore future events. It may be more work because we need to add stop functions for everything, but I think most of those already exist, and it'll make testing things separately simpler.

Reply:
I agree but this PR is already quite large and a preliminary task for another large task.
I'll add a comment here

Reply:
Seems reasonable :)